tests/puppeteer/pages/BO/payment/preferences/index.js

```diff
@@ -35,4 +38,30 @@ module.exports = class Preferences extends BOBasePage {
     await this.page.click(this.currencyRestrictionsSaveButton);
     return this.getTextContent(this.alertSuccessBlock);
   }
+
+  /**
+   * Set group restrictions
+   * @param group
+   * @param paymentModule
+   * @param valueWanted
+   * @returns {Promise<string>}
+   */
+  async setGroupRestrictions(group, paymentModule, valueWanted) {
+    await this.waitForVisibleSelector(`${this.paymentModuleCheckbox
+      .replace('%PAYMENTMODULE', paymentModule)
+      .replace('%GROUPID', group)} + i`,
+    );
+    const isCheckboxSelected = await this.isCheckboxSelected(this.paymentModuleCheckbox
+      .replace('%PAYMENTMODULE', paymentModule)
+      .replace('%GROUPID', group),
+    );
+    if (valueWanted !== isCheckboxSelected) {
+      await this.page.click(`${this.paymentModuleCheckbox
+        .replace('%PAYMENTMODULE', paymentModule)
+        .replace('%GROUPID', group)} + i`,
+      );
+    }
```

Comment:
```suggestion
   const selector = this.paymentModuleCheckbox
      .replace('%PAYMENTMODULE', paymentModule)
      .replace('%GROUPID', group);
    await this.waitForVisibleSelector(`${selector} + i`);
    const isCheckboxSelected = await this.isCheckboxSelected(selector);
    if (valueWanted !== isCheckboxSelected) {
      await this.page.click(`${selector} + i`);
    }
```
Maybe you should check lint errors with this modification.